packages/ppx/src/declarations_to_string.re

```diff
@@ -0,0 +1,425 @@
+  string_to_expr: string => result(list(Parsetree.expression), string),
+};
+
+let emit = (property, value_of_ast, value_to_expr) => {
+  let ast_of_string = Parser.parse(property);
+  let ast_to_expr = ast => value_of_ast(ast) |> value_to_expr;
+  let string_to_expr = string =>
+    ast_of_string(string) |> Result.map(ast_to_expr);
+
+  {ast_of_string, value_of_ast, value_to_expr, ast_to_expr, string_to_expr};
+};
+
+let variants_to_string =
+  fun
+  | `Row => id([%expr "row"])
+  | `Row_reverse => id([%expr "row-reverse"])
+  | `Column => id([%expr "column"])
+  | `Column_reverse => id([%expr "column-reverse"])
+  | `Nowrap => id([%expr "nowrap"])
+  | `Wrap => id([%expr "wrap"])
+  | `Wrap_reverse => id([%expr "wrap-reverse"])
+  | `Content => id([%expr "content"])
+  | `Flex_start => id([%expr "flex-start"])
+  | `Flex_end => id([%expr "flex-end"])
+  | `Center => id([%expr "center"])
+  | `Space_between => id([%expr "space-between"])
+  | `Space_around => id([%expr "space-around"])
+  | `Baseline => id([%expr "baseline"])
+  | `Stretch => id([%expr "strecth"])
+  | `Auto => id([%expr "auto"])
+  | `None => id([%expr "none"])
+  | `Content_box => id([%expr "content-box"])
+  | `Border_box => id([%expr "border-box"])
+  | `Clip => id([%expr "clip"])
+  | `Hidden => id([%expr "hidden"])
+  | `Visible => id([%expr "visible"])
+  | `Scroll => id([%expr "scroll"])
+  | `Ellipsis => id([%expr "ellipsis"])
+  | `Capitalize => id([%expr "capitalize"])
+  | `Lowercase => id([%expr "lowercase"])
+  | `Uppercase => id([%expr "uppercase"])
+  | `Break_spaces => id([%expr "break-spaces"])
+  | `Normal => id([%expr "normal"])
+  | `Break_all => id([%expr "break-all"])
+  | `Break_word => id([%expr "break-word"])
+  | `Keep_all => id([%expr "keep-all"])
+  | `Anywhere => id([%expr "anywhere"])
+  | `BreakWord => id([%expr "breakword"])
+  | `End => id([%expr "end"])
+  | `Justify => id([%expr "justify"])
+  | `Left => id([%expr "left"])
+  | `Match_parent => raise(Unsupported_feature)
+  | `Right => id([%expr "right"])
+  | `Start => id([%expr "start"])
+  | `Transparent => id([%expr "transparent"])
+  | `Bottom => id([%expr "bottom"])
+  | `Top => id([%expr "top"])
+  | `Fill => id([%expr "fill"])
+  | `Dotted => id([%expr "dotted"])
+  | `Dashed => id([%expr "dashlet"])
+  | `Solid => id([%expr "solid"])
+  | `Double => id([%expr "double"])
+  | `Groove => id([%expr "groove"])
+  | `Ridge => id([%expr "ridge"])
+  | `Inset => id([%expr "inset"])
+  | `Outset => id([%expr "outset"])
+  | `Contain => id([%expr "contain"])
+  | `Scale_down => id([%expr "scale-down"])
+  | `Cover => id([%expr "cover"])
+  | `Full_width => raise(Unsupported_feature)
+  | `Unset => id([%expr "unset"])
+  | `Full_size_kana => raise(Unsupported_feature);
+
+let render_number = (number, unit) =>
+  string_of_int(Float.to_int(number)) ++ unit |> render_string;
+let render_percentage = percentage =>
+  "%" ++ string_of_float(percentage *. 100.) |> render_string;
```

Comment:
Is that correct? seems like the wrong order

Reply:
Right, thanks!